Pycloud Deployer/modules.py

```diff
@@ -0,0 +1,277 @@
+        else:
+            return group_name
+        
+        group_name = input("Enter another group name: ")
+
+
+def get_vnet_or_subnet_name(isVNet):
+    """Validate the name of a subnet or VNet
+
+    Args:
+        isVNet (bool): Determine if it is a VNet or Subnet to be validated
+
+    Returns:
+        str: Name of either a Subnet or VNet
+    """
+    # The resource type is VNet if isVNet is True. Else it is set to Subnet.
+    rss_type = "VNet" if isVNet else "Subnet"
+
+    while True:
+        name = input(f"Enter {rss_type} name: ")
+        
+        if not re.match(r'^[a-zA-Z0-9][\w\.-]*[a-zA-Z0-9_]$', name):
+            print(f"Invalid {rss_type} name. Name must begin with a letter or number,\
+                  end with a letter, number, or underscore, and may contain only letters, \
+                  numbers, underscores, periods, or hyphens.")
+        elif not len(name) <= 90:
+            print(f"Invalid {rss_type} name. Name cannot be more than 90 characters.")
+        else:
+            return name
+
+
+def get_nsg_name():
+    """Validates NSG group name
+
+    Returns:
+        str: Name of NSG group
+    """
+    while True:
+        nsg_name = input("Enter NSG name: ")
+
+        if not re.match(r'^[a-zA-Z0-9][a-zA-Z0-9_.-]*[a-zA-Z0-9_]$', nsg_name):
+            print("Invalid NSG name. Name must begin with a letter or number, end \
+                  with a letter, number, or underscore, and may contain only letters, \
+                  numbers, underscores, periods, or hyphens.")
+        elif not 2 <= len(nsg_name) <= 64:
+            print("Invalid NSG name. Name must be between 2 and 64 characters.")
+        else:
+            return nsg_name
+
+
+def get_server_name(isWebServer):
+    """Validates a server's name
+
+    Args:
+        isWebServer (bool): If it is a web or db server name to be validated
+
+    Returns:
+        str: Name of either the web or db server
+    """
+    vm_type = 'Webserver' if isWebServer else 'DBServer'
+
+    while True:
+        vm_name = input(f"Enter {vm_type} name: ")
+        
+        if not re.match(r"^[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]$", vm_name):
+            print(f"Invalid {vm_type} name. Name cannot contain whitespace or special \
+                  characters except '-' but cannot begin or end with '-'")
+        elif not 1 <= len(vm_name) <= 64:
+            print(f"Invalid {vm_type} name. Name must be between 1 and 64 characters.")
+        else:
+            return vm_name
+
+
+def get_username(isWebServer):
+    """Vaidates a login username
+
+    Args:
+        isWebServer (bool): If it is a web or db server name to be validated
+    
+    Returns:
+        str: Login username
+    """
+    server_type = 'webserver' if isWebServer else 'dbserver'
+    reserved_words = ['admin', 'root', 'superuser', "user"]
+
+    while True:
+        username = input(f"Enter a username for the {server_type}: ")
+
+        if not 9 < len(username) < 64:
```

Comment:
The condition not 9 < len(username) < 64 seems incorrect. It should be not (9 < len(username) < 64). Also, the check for reserved words is case-sensitive. You might want to use username.lower() in the comparison to make it case-insensitive.